crates/wasmtime/src/runtime/vm/traphandlers.rs

```diff
@@ -866,8 +867,28 @@ impl CallThreadState {
 
     pub(crate) fn entry_trap_handler(&self) -> Handler {
         unsafe {
+            fn running_on_continuation(stack_chain: *const VMStackChain) -> bool {
+                unsafe {
+                    match *stack_chain {
+                        VMStackChain::Continuation(_) => true,
+                        _ => false,
+                    }
```

Comment:
Should be able to be simplified slightly to just `matches!(*stack_chain, VMStackChain::Continuation(_))`